x/slashing/client/rest/query.go

```diff
@@ -41,65 +40,55 @@ func signingInfoHandlerFn(cliCtx context.CLIContext, storeName string, cdc *code
 			return
 		}
 
-		signingInfo, code, err := getSigningInfo(cliCtx, storeName, cdc, pk.Address())
+		params := slashing.NewQuerySigningInfoParams(sdk.ConsAddress(pk.Address()))
+
+		bz, err := cdc.MarshalJSON(params)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
+			return
+		}
 
+		route := fmt.Sprintf("custom/%s/%s", slashing.QuerierRoute, slashing.QuerySigningInfo)
+		res, err := cliCtx.QueryWithData(route, bz)
 		if err != nil {
-			rest.WriteErrorResponse(w, code, err.Error())
+			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
 			return
 		}
 
-		rest.PostProcessResponse(w, cdc, signingInfo, cliCtx.Indent)
+		rest.PostProcessResponse(w, cdc, res, cliCtx.Indent)
 	}
 }
 
 // http request handler to query signing info
-func signingInfoHandlerListFn(cliCtx context.CLIContext, storeName string, cdc *codec.Codec) http.HandlerFunc {
+func signingInfoHandlerListFn(cliCtx context.CLIContext, cdc *codec.Codec) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		var signingInfoList []slashing.ValidatorSigningInfo
-
 		_, page, limit, err := rest.ParseHTTPArgs(r)
 		if err != nil {
 			rest.WriteErrorResponse(w, http.StatusBadRequest, err.Error())
 			return
 		}
 
-		height, err := rpc.GetChainHeight(cliCtx)
-		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-			return
+		// Override default limit if it wasn't provided as the querier will set
+		// it to the correct default limit.
+		if l := r.FormValue("limit"); l == "" {
+			limit = 0
```

Comment:
A cleaner structure would be to have a function `rest.ParseHTTPArgsWithLimit` (whereby `rest.ParseHTTPArgs` would also reference this new function). Then there would be no explanation req'd!

Reply:
Agreed, I'll update.